### ISSUE TITLE
enrich datapath WireGuard tracing from/to cilium_wg0

### ISSUE THREAD

**Issue by smagnani96** [cilium/cilium#35280]
Follow-up of https://github.com/cilium/cilium/issues/34659.
In the previous issue, we wanted to add logic to mark the encrypted WireGuard packet, as well as for IPSec.
However, this does not provide further useful information from Hubble/Cilium-monitor perspective, as only the outer network packet with nodes IPs is traced with TRACE_REASON_ENCRYPTED.

In this issue we should instead consider additional tracing info from datapath alongside with additional parsing logic from userspace to enrich the capture concerning endpoint communications. It would be useful, for instance, that from Hubble we could see that a specific endpoint flow is encrypted -- even though we trace the plaintext packet.
In this regard, a possible implementation concerning WireGuard could be composed by:
1. always attach `cil_from_wireguard` and `cil_to_wireguard`
2. trace from these hooks using a new  `TRACE_{FROM,TO}_CRYPTO_DEV` reason
3. add cilium-monitor/Hubble logic to correctly display such trace event

**Comment by jschwinger233** [comment]
I think ipsec can also benefit from the two new events, as long as `TRACE_TO_CRYPTO` is defined for flows confirmed to be while yet to be encrypted, `TRACE_FROM_CRYPTO` is for flows already been decrypted.

**Comment by smagnani96** [comment]
> I think ipsec can also benefit from the two new events, as long as `TRACE_TO_CRYPTO` is defined for flows confirmed to be while yet to be encrypted, `TRACE_FROM_CRYPTO` is for flows already been decrypted.

Agree! As you mention, in that case I would just drop the `_DEV` from the name and keep `TRACE_{FROM,TO}_CRYPTO`.
